src/app/components/quiz.tsx

```diff
@@ -0,0 +1,110 @@
+"use client";
+
+import { useState } from "react";
+import clsx from "clsx";
+
+export interface QuizParams {
+  className?: string;
+  question: string;
+  hint?: string;
+  answers: string[];
+  correctAnswer: number;
+  hintAnswers?: string[];
+  onSelect?: (selection: number, isDone: boolean) => void;
+  // If enabled, the user only has a single try. After selection, it will show the solution to the user.
+  showCorrectAnswer?: boolean;
+}
+
+/**
+ * A quiz component.
+ * @param quiz The quiz parameters
+ */
+export default function Quiz(quiz: QuizParams) {
+  const [selection, setSelection] = useState<number>();
```

Comment:
looks suspisious to me. there isn't any default value in the useState????????
typescript is infering the type of the selection if there is a defaultValue in the useState. 

`const [selection, setSelection] = useState(0)` for example

Reply:
It should be undefined by default, however I can make it explicit. Having it undefined makes it easier and more concise to use in conditional rendering